File: JS/Medium/BinaryTree/InOrder/kthSmallest.js
/**230
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */
var kthSmallest = function (root, k) {
    let res = null;

    let inOrder = (curr) => {
        if (!curr || res) return;
        inOrder(curr.left);
        --k;
        if (k == 0) res = curr.val;
        inOrder(curr.right);

    };
    inOrder(root);
    return res;
};

File: C#/DSA/BST/kthSmallest.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solutionk
{
    public int KthSmallest(TreeNode root, int k)
    {
        //Do In order to get node in sorted order
        int ans = 0;
        int count = 0;

        void InOrder(TreeNode root, int k)
        {
            if (root == null || ans != 0) return;
            InOrder(root.left, k);
            count += 1;
            if (k == count)
            {
                ans = root.val;
                return;
            }
            Console.WriteLine("count" + count);
            InOrder(root.right, k);
        }
        InOrder(root, k);
        return ans;
    }


}